### PR TITLE
feat(Video Stitcher): Add samples and tests

### PR DESCRIPTION
**Pull request by irataxy**
b:202972981

Part 2 of 3: VOD sessions. See also https://cloud.google.com/video-stitcher/docs/how-to/creating-vod-sessions and https://cloud.google.com/video-stitcher/docs/how-to/inspecting-vod-sessions

### CODE REVIEW

File: media/stitcher/api/Stitcher.Samples.Tests/ListVodStitchDetailsTests.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+        public ListVodStitchDetailsTest(StitcherFixture fixture)
+        {
+            _fixture = fixture;
+            _createSample = new CreateVodSessionSample();
+            _listSample = new ListVodStitchDetailsSample();
+
+            var result = _createSample.CreateVodSession(
+               _fixture.ProjectId, _fixture.LocationId, _fixture.VodSourceUri, _fixture.VodAdTagUri);
+            _vodSessionId = result.VodSessionName.VodSessionId;
```
Comment (amanda-tarafa) [inline media/stitcher/api/Stitcher.Samples.Tests/ListVodStitchDetailsTests.cs:38]: Does this needs to be deleted?

File: media/stitcher/api/Stitcher.Samples.Tests/GetVodStitchDetailTests.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+
+        private string _vodSessionId;
+        private string _stitchDetailId;
+
+        public GetVodStitchDetailTest(StitcherFixture fixture)
+        {
+            _fixture = fixture;
+            _createSample = new CreateVodSessionSample();
+            _getSample = new GetVodStitchDetailSample();
+            _listSample = new ListVodStitchDetailsSample();
```
Comment (amanda-tarafa) [inline media/stitcher/api/Stitcher.Samples.Tests/GetVodStitchDetailTests.cs:37]: I'm guessing no, but can't we have one VodSession with it's detail created by the fixture so we can use them on the get and list tests?

File: media/stitcher/api/Stitcher.Samples/CreateVodSession.cs [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    public VodSession CreateVodSession(
+        string projectId, string location, string sourceUri, string adTagUri)
+    {
+        // Create the client.
+        VideoStitcherServiceClient client = VideoStitcherServiceClient.Create();
+
+        CreateVodSessionRequest request = new CreateVodSessionRequest
+        {
+            ParentAsLocationName = LocationName.FromProjectLocation(projectId, location),
+            VodSession = new VodSession()
```
Comment (amanda-tarafa) [inline media/stitcher/api/Stitcher.Samples/CreateVodSession.cs:33]: super tiny nit:
```suggestion
            VodSession = new VodSession

```

**Review by amanda-tarafa — COMMENTED** [review]
LGTM with a general question:

Can we not create the VodSessions on the fixture so they can be reused by the Get and List tests? Also, are there quotas for these? Should they be deleted?

Thanks!

File: media/stitcher/api/Stitcher.Samples.Tests/ListVodStitchDetailsTests.cs (hunk 1 above)
Comment (irataxy) [reply media/stitcher/api/Stitcher.Samples.Tests/ListVodStitchDetailsTests.cs:38]: There is no delete method for a VOD session (no quota either). The session expires after 4 hours.

File: media/stitcher/api/Stitcher.Samples.Tests/GetVodStitchDetailTests.cs (hunk 2 above)
Comment (irataxy) [reply media/stitcher/api/Stitcher.Samples.Tests/GetVodStitchDetailTests.cs:37]: We can but there is no quota or delete method for a VOD session, so if it's ok, I'd like to keep the create code in the test files.

**Review by amanda-tarafa — APPROVED** [review]
LGTM

**Pull request merged**